[CORE-721] Add start and end date query params to list submissions endpoint

samanehsan opened the pull request:
Ticket: https://broadworkbench.atlassian.net/browse/CORE-721

Update listSubmissions endpoint to take in optional `startDate` and `endDate` query parameters:
- If no query params are specified, default to the existing behavior which is to list all submissions. 
- If a start and date are specified, return results within that date range
- If only a start date is specified, return all results after (and including) that date
- If only an end date is specified, return all results before (and including) that date  

---

**PR checklist**

- [x] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should publish a new official `rawls-model` and perform the corresponding dependency updates as specified in the [README](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model):
  - [ ] in the automation subdirectory
  - [ ] in workbench-libs
  - [ ] in firecloud-orchestration
- [x] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

samanehsan committed:
Add start and end date query params to list submissions API endpoint

samanehsan committed:
Add tests

samanehsan committed:
Scalafmt and updating swagger example

samanehsan committed:
Remove commented code

samanehsan committed:
Fix unit test failure

davidangb commented on core/src/main/resources/swagger/api-docs.yaml:
```diff
@@ -746,6 +746,21 @@ paths:
+          schema:
+            type: string
+            format: date
+            example: 2025-10-31
```
If the user omits these, we need to default to returning all submissions i.e. no date filtering. Otherwise, we're changing behavior of this API for scripters, etc. and we decided we don't want to do that.

samanehsan replied:
Oh you're right! I mixed this up with the UI defaulting to the last 30 days

davidangb replied:
Ah I see you added these query params to `/api/workspaces/{workspaceNamespace}/{workspaceName}/submissions` below ... they should get removed from `/api/admin/submissions` here, since we haven't updated that endpoint.

samanehsan replied:
I think it would be nice to have that for consistency, so I'll make a follow-on ticket for it

davidangb reviewed (changes requested):
code looks good; one request for a behavior change

samanehsan committed:
Merge branch 'develop' into se/CORE-721-filter-submissions-by-date

samanehsan committed:
List all submissions when no date filters are specified

samanehsan committed:
Merge branch 'se/CORE-721-filter-submissions-by-date' of github.com:broadinstitute/rawls into se/CORE-721-filter-submissions-by-date

samanehsan committed:
Update tests

samanehsan committed:
Updated submissions filtering behavior

- If only a start date is specified, return all results
  after that date (inclusive)
- If only an end date is specified, return all results
  before that date (inclusive)

davidangb commented on core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala:
```diff
@@ -151,11 +151,25 @@ trait SubmissionComponent {
-    def listWithSubmitter(workspaceContext: Workspace): ReadWriteAction[Seq[SubmissionListResponse]] = {
+    def listWithSubmitter(workspaceContext: Workspace,
+                          startDateOpt: Option[DateTime],
+                          endDateOpt: Option[DateTime]
+    ): ReadWriteAction[Seq[SubmissionListResponse]] = {
+      val baseQuery = findByWorkspaceId(workspaceContext.workspaceIdAsUUID)
+      val filteredQuery = (startDateOpt, endDateOpt) match {
+        case (Some(startDate), Some(endDate)) =>
+          val start = new Timestamp(startDate.withTimeAtStartOfDay.getMillis)
+          val end = new Timestamp(endDate.plusDays(1).withTimeAtStartOfDay.getMillis - 1)
+          baseQuery.filter(sub =>
+            sub.submissionDate >= start && sub.submissionDate <= end
+          )
+        case _ =>
+          baseQuery
```
Could you apply the start filter and the end filter independently? That is, if user only supplies one of the query params, that query param will still take effect?

In that case, "last 30 days" would only need to specify the start date, and can omit the end date.

samanehsan replied:
Yes! I just made that change

samanehsan committed:
Fix test failures

samanehsan commented on core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/DriverComponentSpec.scala:
```diff
@@ -73,7 +73,7 @@ class DriverComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers wit
 
     // first check that we're not just comparing empty seqs
-    assertResult(33)(queryRecords.length)
+    assertResult(34)(queryRecords.length)
```
This is from adding another test submission with a workflow

samanehsan committed:
Add date filters to workspace submission list endpoint

samanehsan committed:
Fix tests

samanehsan committed:
Fix list submissions admin endpoint definition

davidangb reviewed (approved):
thank you!

samanehsan commented:
The flakey tests keep failing 😞
